tsfc/driver.py

```diff
@@ -267,21 +269,104 @@ def name_multiindex(multiindex, name):
     return builder.construct_kernel(kernel_name, impero_c, parameters["precision"], index_names, quad_rule)
 
 
-def compile_expression_at_points(expression, points, coordinates, interface=None,
-                                 parameters=None, coffee=True):
-    """Compiles a UFL expression to be evaluated at compile-time known
-    reference points.  Useful for interpolating UFL expressions onto
-    function spaces with only point evaluation nodes.
+def apply_mapping(expression, to_element):
```

Comment:
It wasn't immediately obvious for me that `expression` is UFL expression, and `to_element` is FIAT element...  However, `to_element` is only used to query the mapping, so I suggest the following slight change: pass in `mapping` string instead of FIAT element (so FIAT element -> mapping string code moves to the caller), then this function will depend on UFL only, and can be moved to `ufl_utils.py` where the other UFL processors are.

In `ufl_utils.py` you can also import Index and MultiIndex from `ufl.classes`, so no need to use fully-qualified names.

Reply:
Good idea! Both done.